build.gradle

```diff
@@ -0,0 +1,176 @@
+
+plugins {
+    id "java"
+    id 'maven'
+    id 'signing'
+    id 'jacoco'
+    id 'com.palantir.git-version' version '0.5.1' //version helper
+}
+
+group 'barclay'
+version '1.0-SNAPSHOT'
+
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
+group = 'org.broadinstitute'
+
+final isRelease = Boolean.getBoolean("release")
+version = (isRelease ? gitVersion() : gitVersion() + "-SNAPSHOT").replaceAll(".dirty", "")
+
+repositories {
+    mavenCentral()
+    maven {
+        url "https://artifactory.broadinstitute.org/artifactory/libs-snapshot/" //for htsjdk snapshots
+    }
+
+    mavenLocal()
+}
+
+jacocoTestReport {
+    dependsOn test
+    group = "Reporting"
+    description = "Generate Jacoco coverage reports after running tests."
+    additionalSourceDirs = files(sourceSets.main.allJava.srcDirs)
+
+    reports {
+        xml.enabled = true // coveralls plugin depends on xml format report
+        html.enabled = true
+    }
+}
+
+compileJava {
+    options.compilerArgs = ['-proc:none', '-Xlint:all','-Werror','-Xdiags:verbose']
+}
+compileTestJava {
+    options.compilerArgs = ['-proc:none', '-Xlint:all','-Werror','-Xdiags:verbose']
+}
+
+dependencies {
+    compile 'com.google.guava:guava:18.0'
+
+    compile 'net.sf.jopt-simple:jopt-simple:5.0-beta-1'
+
+    compile 'org.apache.commons:commons-lang3:3.4'
```

Comment:
I'm afraid to see... `com.sun` private dependencies everywhere?  It would be ideal to have separate artifacts if we end up having a lot of docgen only dependencies.  Not great to have to package those in our final jars, especially any of the jvm specific tool jars.